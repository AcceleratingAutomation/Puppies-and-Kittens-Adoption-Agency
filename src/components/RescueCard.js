import React from "react";
import { Grid, Typography, IconButton } from "@material-ui/core";
import FavoriteIcon from '@material-ui/icons/Favorite';
import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder';
import "../styles.css";
import { SummaryCard } from "./SummaryCard";

export const RescueCard = ({ name, id, type, gender, breed, image, isFavorite, onAddFavorite, onRemoveFavorite }) => {
  const rescue = { name, id, type, gender, breed, image };

  return (
    <SummaryCard type={rescue.type} image={rescue.image} name={rescue.name} viewComponentDetailsUrl={`/v1/rescueDetails/${rescue.id}`}>
      <Grid item xs={12}>
        <Typography variant="h4">{rescue.name}</Typography>
      </Grid>
      <Typography variant="h5" gutterBottom>
        {rescue.type}
      </Typography>
      <Typography variant="h5" gutterBottom>
        {rescue.gender}
      </Typography>
      <Typography variant="h5" gutterBottom>
        {rescue.breed}
      </Typography>
      <Grid item xs={12} container justify="center">
        {isFavorite ? (
          <IconButton
            color="secondary"
            onClick={() => onRemoveFavorite(rescue.id)}
          >
            <FavoriteIcon />
          </IconButton>
        ) : (
          <IconButton
            onClick={() => onAddFavorite(rescue.id)}
          >
            <FavoriteBorderIcon />
          </IconButton>
        )}
      </Grid>
    </SummaryCard>
  );
};