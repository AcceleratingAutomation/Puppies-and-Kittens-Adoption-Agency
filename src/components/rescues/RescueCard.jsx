import React from "react";
import { Grid, Typography, IconButton } from "@mui/material";
import FavoriteIcon from "@mui/icons-material/Favorite";
import FavoriteBorderIcon from "@mui/icons-material/FavoriteBorder";
import "../../styles.css";
import PropTypes from "prop-types";
import SummaryCard from "../SummaryCard";
import { rescuesEndpoint } from "../../server/apiService/apiConfig";

function RescueCard({
  name,
  id,
  image,
  type,
  gender,
  breed,
  isFavorite,
  onAddFavorite,
  onRemoveFavorite,
}) {
  const rescue = {
    name,
    id,
    image,
    type,
    gender,
    breed,
  };

  return (
    <SummaryCard
      directory={rescue.type.toLowerCase()}
      type={type}
      id={rescue.id}
      image={rescue.image}
      name={rescue.name}
      viewComponentDetailsUrl={`${rescuesEndpoint}/${rescue.id}`}
    >
      <Grid item xs={12}>
        <Typography variant="h4">{rescue.name}</Typography>
      </Grid>
      <Typography variant="h5" gutterBottom>
        {rescue.type}
      </Typography>
      <Typography variant="h5" gutterBottom>
        {rescue.gender}
      </Typography>
      <Typography variant="h5" gutterBottom>
        {rescue.breed}
      </Typography>
      <Grid item xs={12} container justifyContent="center">
        {isFavorite ? (
          <IconButton
            color="error"
            onClick={() => onRemoveFavorite(rescue.id)}
            aria-label="Remove from favorites"
          >
            <FavoriteIcon />
          </IconButton>
        ) : (
          <IconButton
            onClick={() => onAddFavorite(rescue.id)}
            aria-label="Add to favorites"
          >
            <FavoriteBorderIcon />
          </IconButton>
        )}
      </Grid>
    </SummaryCard>
  );
}

RescueCard.defaultProps = {
  onAddFavorite: () => {}, // TODO
};

RescueCard.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  image: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  type: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  breed: PropTypes.string.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  onAddFavorite: PropTypes.func,
  onRemoveFavorite: PropTypes.func.isRequired,
};

export default RescueCard;
