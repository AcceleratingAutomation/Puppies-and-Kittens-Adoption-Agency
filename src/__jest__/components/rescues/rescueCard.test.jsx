import React from "react";
import { render, screen } from "@testing-library/react";
import RescueCard from "../../../components/rescues/RescueCard";

function renderRescueCard(props) {
  return render(
    <RescueCard
      id={props.id}
      name={props.name}
      image={props.image}
      type={props.type}
      gender={props.gender}
      breed={props.breed}
      isFavorite={props.isFavorite}
      onAddFavorite={props.onAddFavorite}
      onRemoveFavorite={props.onRemoveFavorite}
    />,
  );
}

describe("Rescue Card", () => {
  const rescueProps = {
    id: "1",
    name: "Max",
    image: "thumbnail-image",
    type: "Dog",
    gender: "Male",
    breed: "Labrador",
    isFavorite: false,
    onAddFavorite: jest.fn(),
    onRemoveFavorite: jest.fn(),
  };

  beforeEach(() => {
    renderRescueCard(rescueProps);
  });

  test("matches snapshot comparison", () => {
    const { asFragment } = renderRescueCard(rescueProps);
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays rescue name", () => {
    expect(screen.getByText(rescueProps.name)).toBeInTheDocument();
  });

  test("displays type as Dog", () => {
    expect(screen.getByText(rescueProps.type)).toBeInTheDocument();
  });

  test("displays type as Cat", () => {
    const catProps = {
      ...rescueProps,
      type: "Cat",
    };
    renderRescueCard(catProps);
    expect(screen.getByText(catProps.type)).toBeInTheDocument();
  });

  test("displays gender as Male", () => {
    expect(screen.getByText(rescueProps.gender)).toBeInTheDocument();
  });

  test("displays gender as Female", () => {
    const femaleProps = {
      ...rescueProps,
      gender: "Female",
    };
    renderRescueCard(femaleProps);
    expect(screen.getByText(femaleProps.gender)).toBeInTheDocument();
  });

  test("displays breed", () => {
    expect(screen.getByText(rescueProps.breed)).toBeInTheDocument();
  });

  test("displays favorite status as not a favorite", () => {
    const favoriteButton = rescueProps.isFavorite
      ? screen.getByLabelText("Remove from favorites")
      : screen.getByLabelText("Add to favorites");
    expect(favoriteButton).toBeInTheDocument();
  });

  test("displays favorite status as a favorite", () => {
    const favoriteProps = {
      ...rescueProps,
      isFavorite: true,
    };
    renderRescueCard(favoriteProps);
    const favoriteButton = screen.getByLabelText("Remove from favorites");
    expect(favoriteButton).toBeInTheDocument();
  });
});
