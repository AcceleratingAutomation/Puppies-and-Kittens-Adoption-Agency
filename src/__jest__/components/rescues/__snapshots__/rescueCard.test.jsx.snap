// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rescue Card matches snapshot comparison 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root rescue makeStyles-summaryCard-1 MuiPaper-elevation2 MuiPaper-rounded"
  >
    <img
      alt="Max's picture"
      class="makeStyles-responsiveImage-3"
      src="/images/dog/thumbnail-image.webp"
      style="border-radius: 25%; width: 15rem; height: 18rem; object-fit: cover;"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          Max
        </h4>
      </div>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Dog
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Male
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Labrador
      </h5>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
      >
        <button
          aria-label="Add to favorites"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-muiButton-2 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            VIEW DETAILS
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
